google/cloud/security/scanner/scanners/groups_scanner.py

```diff
@@ -0,0 +1,168 @@
+        """
+        all_violations = []
+        for node in iterators.PreOrderIter(root):
+
+            # No need to evaluate these nodes.
+            # This represents the org, i.e. is not a group.
+            if node.member_email == 'my_customer':
+                continue
+            # This represents the auto-generated group, containing all the users
+            # in the org.
+            if node.member_email == '':
+                continue
+
+            node.violations = []
+            whitelist_rule_statuses = []
+            for rule in node.rules:
+                condition_statuses = []
+
+                if rule.get('mode') == 'whitelist':
+                    for condition in rule.get('conditions'):
+                        if condition.get('member_email') in node.member_email:
+                            condition_statuses.append(True)
+                        else:
+                            condition_statuses.append(False)
+
+                    # All the conditions of this rule have evaluated.
+                    # The rule is fulfilled, if any condition matches.
+                    if any(condition_statuses) is True:
+                        whitelist_rule_statuses.append(True)
+                    else:
+                        whitelist_rule_statuses.append(False)
+
+                elif rule.get('mode') == 'blacklist':
+                    pass  # TODO
+
+                elif rule.get('mode') == 'required':
+                    pass  # TODO
+
+                else:
+                    pass  # TODO
+    
+            # Determine if the node is in violations or not.
+            # All rules must be fulfilled, for a node to not be in violation.
+            # If any rule is not fulfilled, then node is in violation.
+            #
+            # truth table
+            # http://stackoverflow.com/a/19389957/2830207
+            # +-----------------------------------------+---------+---------+
+            # |                                         |   any   |   all   |
+            # +-----------------------------------------+---------+---------+
+            # | All Truthy values                       |  True   |  True   |
+            # +-----------------------------------------+---------+---------+
+            # | All Falsy values                        |  False  |  False  |
+            # +-----------------------------------------+---------+---------+
+            # | One Truthy value (all others are Falsy) |  True   |  False  |
+            # +-----------------------------------------+---------+---------+
+            # | One Falsy value (all others are Truthy) |  True   |  False  |
+            # +-----------------------------------------+---------+---------+
+            # | Empty Iterable                          |  False  |  True   |
+            # +-----------------------------------------+---------+---------+
+            if any(whitelist_rule_statuses) is False:
```

Comment:
Is there a reason you chose "if any(...) is False" as opposed to "if not any(...)" ?